### PR TITLE
ListOpnd for lowered IR

### PR DESCRIPTION
**Pull request by Cellule**
Create new Opnd ListOpnd. 
Currently available only at lower and after.
Allows to set a list of dst and src.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/3978)

### CODE REVIEW

**Comment by Cellule** [comment]
@Microsoft/chakra-jit FYI
@agarwal-sandeep & @sigatrev , I believe you guys could use this for arm64.

**Comment by Cellule** [comment]
I have tested the behavior in my change for WebAssembly Threads, but I am splitting that change into multiple PRs.

File: lib/Backend/Opnd.cpp [hunk 1]
```diff
@@ -2288,6 +2304,110 @@ AddrOpnd::SetAddress(Js::Var address, AddrOpndKind addrOpndKind)
...
+        }
+    }
+    return true;
+}
+
+Opnd * ListOpnd::CloneUseInternal(Func * func)
+{
+    Assert(m_kind == OpndKindList);
+    int count = Count();
+    ListOpndType** opnds = JitAnewArray(func->m_alloc, RegOpnd*, count);
```
Comment (rajatd) [inline lib/Backend/Opnd.cpp]: >RegOpnd [](start = 55, length = 7)

Use ListOpndType for consistency? (here and below)

File: lib/Backend/Opnd.h [hunk 2]
```diff
@@ -1467,6 +1472,105 @@ class AddrOpnd sealed : public Opnd
...
+    template<typename K1, typename... K>
+    void insert(int index, K1 arg, K... rest)
+    {
+        values[index] = arg;
+        insert(index + 1, rest...);
+    }
+    template<typename K>
+    void insert(int index, K last)
+    {
+        values[index] = last;
```
Comment (agarwal-sandeep) [inline lib/Backend/Opnd.h]: >values[index] = last; [](start = 8, length = 21)

Is overwriting allowed?

Comment (Cellule) [reply lib/Backend/Opnd.h]: No, this structure should be a type safe variable arguments buffer that can be optimized away at compile time.
@penguinwizzard originally wrote this struct.
I was thinking of making this generic for possibly other purposes.
Also, you're making me realize that the insert method should probably be private

Comment (Penguinwizzard) [reply lib/Backend/Opnd.h]: You could probably even go one step further and make ListOpndInit a private class inside ListOpnd.

**Pull request merged**